### ISSUE TITLE
iOS heavy pdf rendering issue

### ISSUE THREAD

**Issue by airman5573** [wojtekmaj/react-pdf#1926]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [X] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

iOS PDF Rendering Issue with Large Files

We're experiencing PDF rendering issues specifically on iOS devices. The browser becomes unresponsive when trying to render certain PDF files.

### Steps to reproduce

With iOS phone.

1. go to https://www.modr.co.kr/journal/fola-v1n2/
2. click the '자세히 보기' button 
<img width="896" alt="image" src="https://github.com/user-attachments/assets/826f9f71-38b4-4e67-8a7a-076ae4a73683" />
3. you can see the react-pdf works well.
<img width="1091" alt="image" src="https://github.com/user-attachments/assets/2b9bbb9a-8b74-484b-b4cd-62826c4b821b" />

4. go to https://www.modr.co.kr/journal/tg-gallery-v1n1/
5. click the '자세히 보기' button.
6. you can see the browser is broken.
<img width="396" alt="image" src="https://github.com/user-attachments/assets/56fab4f2-a25e-4825-9f62-47b641e71c35" />

### Expected behavior

I hope large pdf file would open in ios also.

### Actual behavior

browser is broken with large pdf file.

### Environment

- **Browser (if applicable)**: safari latest version
- **React-PDF version**: 9.1.0
- **React version**: 18.3.1
- **Bundler name and version (if applicable)**: vite5

**Comment by vigilantee** [comment]
This is due to the large size of the pdf.
https://github.com/react-pdf-viewer/react-pdf-viewer/issues/490
It is happening on iphone 13 with ios version 18+.
We have replaced the library with adobe. (https://developer.adobe.com/document-services/docs/overview/pdf-embed-api/releasenotes/)

This issue is not going to be fixed in next 15 days as it is end of the year and most of the community will be away from there cubicles.

Adobe is free to use and easy to plug and play.
Here is the reference 


`import React, { useEffect } from "react";
import "./PdfViewer.css";

const clientId = process.env.REACT_APP_ADOBE_CLIENT_ID;

const previewConfig = {
    embedMode: "FULL_WINDOW",
    showDownloadPDF: false,
    showPrintPDF: false,
    enableFormFilling: false,
    showAnnotationTools: false,
    showBookmarks: false,
    showThumbnails: false,
} 

const PdfViewerIos: React.FC<{
  url?: string;
}> = ({
  url = "https://cdn.filestackcontent.com/wcrjf9qPTCKXV3hMXDwK",
}) => {
  useEffect(() => {
    loadDoc();
  }, []);

  const loadDoc = () => {
    // @ts-expect-error This does not exist outside of polyfill which this is doing
    var adobeDCView = new AdobeDC.View({
      clientId,
      divId: "pdf-viewer",
    });
    adobeDCView.previewFile({
      content: {
        location: {
          url,
        },
      },
      metaData: { fileName: "agreement.pdf" },
    }, previewConfig);
  };
  console.log('clientId....', clientId);

  return <div id="pdf-viewer" className="pdf-viewer"></div>;
};

export default PdfViewerIos;
`

**Comment by airman5573** [comment]
Thank you so much for the detailed explanation and background information. I really appreciate it! By any chance, will this issue be fixed after 15 days? We’re not in a huge rush and can wait a little longer. Ideally, we’d prefer to continue using this library if possible.

**Comment by vigilantee** [comment]
what we need is a lazy load feature for the pdf on scroll as that is the only way to reduce the burst of RAM usage(crossing 384mb +) but that will also need the meta data to be maintained to allow the search feature since this feature is bit time taking to implement and there is also a possiblity that iOS fixes this on there end and release the fixes in iOS 18.2 or 18.3.

Difficult to assume this issue to be fixed soon in this library as this will need a very significant development.

**Comment by wojtekmaj** [comment]
See #1020, and https://github.com/wojtekmaj/react-pdf/issues/1020#issuecomment-1323540221 in particular.

**Comment by danielcavanagh** [comment]
ok, good to know where the project stands and some possible workarounds

however, are you sure this is the same issue? with this one there's no warning in the console log, unlike #1020, and there's a hard page crash that isn't manageable with js (as the entire page stops running)

perhaps this is ultimately also a resource limit issue and the outcome will be the same (ie. no fix forthcoming) but it definitely seems like a distinct issue that i haven't seen address anywhere

**Comment by voltabhi** [comment]
> @vigilantee are you sure this is the same issue as [react-pdf-viewer/react-pdf-viewer#490](https://github.com/react-pdf-viewer/react-pdf-viewer/issues/490)?
> 
> that issue seems to at least continue to render the page (albeit with an empty canvas and potentially an exception in js), and it leaves this message in the console:
> 
> ```
> [Warning] Total canvas memory use exceeds the maximum limit (384 MB). (2.c9af885f.chunk.js, line 2)
> ```
> 
> this issue (if it matches my issues, as it seems to) leaves no messages in the console, and it causes a hard page crash that you can't handle with js

@danielcavanagh  You are right this is resulting in hard crash and not in soft crash although the root cause is similar for both of them as mentioned by @wojtekmaj .
